app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -121,7 +167,38 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
-      logger.d("StateFragment", "getCurrentState: ${result.state.name}")
+      if (result.hasState()) {
```

Comment:
State is always present but because this method `hasState()` was public i though maybe use this just to be extra-sure.

Reply:
That shouldn't be needed. Let's remove it in that case--I prefer not defensively programming except in cases where code needs to be made extra robust. In this case, I'd rather see quick failure to stabilize the code.

State should always be present, and if it ever isn't we should see things break so that we can fix that error in the domain layer.

Reply:
Sounds Good. Will revert and resolve this conversation later on.

Reply:
Removed if condition check.